File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length==0){
            return 0;
        }
        int low = prices[0];
        int profit = 0;
        for(int i=0;i<prices.length;i++){
            if(prices[i]<low){
                low = prices[i];
            }
            if(prices[i]-low>profit){
                profit=prices[i]-low;
            }
        }
        return profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if(len(prices)==0):
            return 0
        low = prices[0]
        profit = 0
        for i in range(len(prices)):
            if prices[i]<low:
                low = prices[i]
            
            if((prices[i]-low)>profit):
                profit =prices[i]-low
                
        return profit
            
           